staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/dynamic_cafile_content.go

```diff
@@ -74,7 +74,14 @@ type caBundleAndVerifier struct {
 }
 
 // NewDynamicCAContentFromFile returns a CAContentProvider based on a filename that automatically reloads content
+//
+//logcheck:context // NewDynamicCAContentFromFileWithContext should be used instead because ret.loadCABundle is called and works better when a context is supplied (contextual logging, cancellation).
 func NewDynamicCAContentFromFile(purpose, filename string) (*DynamicFileCAContent, error) {
+	return NewDynamicCAContentFromFileWithContext(context.Background(), purpose, filename)
+}
+
+// NewDynamicCAContentFromFileWithContext returns a CAContentProvider based on a filename that automatically reloads content
+func NewDynamicCAContentFromFileWithContext(ctx context.Context, purpose, filename string) (*DynamicFileCAContent, error) {
```

Comment:
`logger klog.Logger` is probably a better parameter for a `New` method. This applies to all methods in this PR which only use the context to retrieve a logger.